### PR TITLE
Add embed::rc_servo

### LINKED ISSUE #279: Add generic `embed::rc_servo`
Create a generic RC servo driver.

### PR DESCRIPTION
**Pull request by MaliaLabor**
Resolves #279

### CODE REVIEW

File: include/libembeddedhal/servo/rc.hpp [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    , m_max_percent(p_max_percent)
+  {
+  }
+
+  boost::leaf::result<void> driver_position(
+    percent p_position) noexcept override
+  {
+    auto scaled_percent_raw = BOOST_LEAF_CHECK(
+      map(p_position.raw_value() / 2,
+          { .x = percent::raw_min() / 2, .y = percent::raw_max() / 2 },
```
Comment (kammce) [inline include/libembeddedhal/servo/rc.hpp:76]: Why is this divided by 2?

File: tests/servo/rc.test.cpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    expect(bool{ result1 });
+    expect(bool{ result2 });
+    expect(bool{ result3 });
+    expect(bool{ result4 });
+
+    expect(expected0 == std::get<0>(pwm.spy_duty_cycle.call_history().at(0)));
+    expect(expected1 == std::get<0>(pwm.spy_duty_cycle.call_history().at(1)));
+    expect(expected2 == std::get<0>(pwm.spy_duty_cycle.call_history().at(2)));
+    expect(expected3 == std::get<0>(pwm.spy_duty_cycle.call_history().at(3)));
+    expect(expected4 == std::get<0>(pwm.spy_duty_cycle.call_history().at(4)));
```
Comment (kammce) [inline tests/servo/rc.test.cpp]: Can you place `that %` in front of these so we can see the results if this fails? Add the `#include "ostreams.hpp"` to get the code needed for `that` to work.

File: tests/servo/rc.test.cpp [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    embed::mock::pwm pwm0;
+    embed::mock::pwm pwm1;
+
+    // Exercise
+    // use defaults
+    auto servo0 = embed::rc_servo::create(pwm0);
+    // 100Hz (or 10ms per update) with 500us being max negative start and 2500us
+    // being max positive.
+    auto servo1 = embed::rc_servo::create<100, 500, 2500>(pwm1);
+
```
Comment (kammce) [inline tests/servo/rc.test.cpp]: Can you make a test for when create fails for example when configure also fails.

**Review by kammce — CHANGES_REQUESTED** [review]

File: include/libembeddedhal/servo/rc.hpp (hunk 1 above)
Comment (MaliaLabor) [reply include/libembeddedhal/servo/rc.hpp:76]: I was getting an overflow issue when I used the full range

**Comment by kammce** [comment]
Please resolve the messages if they are fixed

File: include/libembeddedhal/servo/rc.hpp (hunk 1 above)
Comment (kammce) [reply include/libembeddedhal/servo/rc.hpp:76]: Yikes, well this is ok for now.

**Review by kammce — APPROVED** [review]

**Pull request merged**